src/glimpse.cpp

```diff
@@ -95,16 +102,16 @@ int main(int argc, char *argv[])
     } catch (po::error &e) {
         std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
         std::cerr << cmdline_options << std::endl;
-        return 1;
+        exit(1);
```

Comment:
What's the rationale of these `return` -> `exit` changes?  To make the program exit immediately?  I think a cleaner approach would be to make the function `int` and check the returned value in the calling site.

Reply:
These were returns from main in the original code, so the program *did* exit immediately at these points.

If we make the `create_config()`function return an int, a return value of 0, 1 or -1 means stop execution and return to caller, for various conditions. What value do we return to communicate, 'all is well, carry on'?

Reply:
> These were returns from main in the original code, so the program did exit immediately at these points.

I understand this, but I'm not 100% convinced that a non-`main` function should kill the program.  I prefer controlling this in the `main`.

> If we make the `create_config()`function return an int, a return value of 0, 1 or -1 means stop execution and return to caller, for various conditions. What value do we return to communicate, 'all is well, carry on'?

0?  We can revise these exit codes anyway.  For example, usually there is no such thing as a negative exit code, -1 is actually 255 (modulo 256 is applied by the shell):
```bash
% bash -c "exit -1"; echo $?          
255
```